src/bedrock/hash_ring.cpp

```diff
@@ -0,0 +1,212 @@
+#include <atomic>
+#include <string>
+#include <functional>
+#include "communication.pb.h"
+#include "zmq/socket_cache.h"
+#include "zmq/zmq_util.h"
+
+#include "spdlog/spdlog.h"
+#include "yaml-cpp/yaml.h"
+
+#include "threads.h"
+#include "requests.h"
+#include "hash_ring.h"
+
+bool operator==(const server_thread_t& l, const server_thread_t& r) {
```

Comment:
Shouldn't this be in `threads.h`?

Reply:
Can't have a function declaration in a .h file or you get multiply-defined errors. I tried making it be an equality member method of server_thread but that broke some template magic elsewhere.

Reply:
Could try this as an inline function in a header file .. not sure if the template magic will break.

Reply:
inlining this seems to work fine!